core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,1002 @@
+
+from core.domain import exp_fetchers
+from core.domain import html_cleaner
+from core.domain import state_domain
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import exp_models
+
+datastore_services = models.Registry.import_datastore_services()
+
+(exp_models,) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class ExpStateValidationJob(base_jobs.JobBase):
+    """Job that tests the general state, rte and interaction validation"""
+
+    @staticmethod
+    def filter_invalid_state_rte_values(
+        states_dict: dict[str, state_domain.State]
+    ) -> list[dict[str, list]]:
+        """Returns the errored state RTE values, validates the following -
+
+        - Image tags contain filepath, alt, and caption attributes,
+        where caption can be an empty string with at most 160
+        characters and alt should have at least 5 characters.
+        - Math tags contain math_content, raw_latex, and svg_filename
+        attributes, where svg_filename has an SVG extension.
+        - Skillreview tags contain text attributes, text is non-empty.
+        - Video tags contain video_id, start, end, and autoplay attributes,
+        where start is before end.
+        - Link tags contain text and url attributes, where text is non-empty
+        and url begins with HTTPS.
+
+        Args:
+            states_dict: dict[str, State]. The dictionary containing
+                state name as key and State object as value.
+
+        Returns:
+            states_with_values: list[dict]. The list of dictionaries
+            containing the errored values.
+        """
+        states_with_values = []
+        for key, value in states_dict.items():
+            rte_components_errors = []
+            rte_components = html_cleaner.get_rte_components(
+                value.content.html)
+            for rte_component in rte_components:
+                # RTE image validations for caption, alt and filepath.
+                if rte_component['id'] == 'oppia-noninteractive-image':
+                    # Validates if caption value is greater than 160.
+                    cap_with_value = rte_component['customization_args'][
+                        'caption-with-value']
+                    alt_with_value = rte_component['customization_args'][
+                        'alt-with-value']
+                    file_with_value = rte_component['customization_args'][
+                        'filepath-with-value']
+
+                    if len(cap_with_value) > 160:
+                        rte_components_errors.append(
+                            'State - {key} Image tag caption value '.format(
+                                key=key) +
+                            'is greater than 160.')
```

Comment:
Don't split the formating like this. Either use f-strings or % formatting

```suggestion
                            f'State - {key} Image tag caption value '
                            f'is greater than 160.'
                        )
```

```suggestion
                            'State - %s Image tag caption value '
                            'is greater than 160.' % key
                        )
```

ditto elsewhere